### DISCUSSION TITLE
How can i debug typescript with ts-node?

### DISCUSSION THREAD

**Discussion by Nirodhiko** [mfussenegger/nvim-dap#659] [Q&A] [answered]
It took me three days to try but I still couldn't debug typescript with ts-node. 
There is very little documentation about [typescript debug configuration](https://twitter.com/ThePrimeagen/status/1538960560355692544).

Has anyone tried it? Can you share the configuration file to help newbies like me？

It's so easy to do so in VS-Code.
I wonder if i should stick with vim, although I have spent a lot of time to learn it.

**Comment by Nirodhiko** [comment, marked as answer]
After a long time of trying, it finally works. 
You can debug typescript file directly with `ts-node`.

This is my configuration：
``` 
  -- set console for pure typescript program (e.g. cli program).
  dap.defaults.fallback.external_terminal = {
    command = "/usr/bin/alacritty",
    args = { "-e" },
  }
  dap.defaults.fallback.force_external_terminal = true
  dap.defaults.fallback.terminal_win_cmd = "50vsplit new"
  dap.defaults.fallback.focus_terminal = true
  dap.set_log_level("INFO")

-- depending on your installation location.
local debugger_location = os.getenv("HOME") .. "/Software/vscode-node-debug2"

dap.adapters.node = {
    type = "executable",
    command = "node",
    args = { debugger_location .. "/out/src/nodeDebug.js" },
 }

dap.configurations.typescript = {
  {
    name = "Lanunch Node",
    type = "node",
    request = "launch",
    runtimeArgs = { "-r", "ts-node/register" },
    runtimeExecutable = "node",
    args = { "--inspect", "${file}" },
    skipFiles = { "node_modules/**" },
    console = "integratedTerminal",
  },
}
```

**Reply by minhtranin** [reply]
I got an issue `Cannot find module 'src/....',` It might be missing require tsconfig-paths, is there any solution?

**Reply by Nirodhiko** [reply]
> I got an issue `Cannot find module 'src/....',` It might be missing require tsconfig-paths, is there any solution?

You have to download vscode-node-debug2.

**Reply by fronterior** [reply]
Thanks.
I'm using it by clearing `runtimeArgs` and replacing `runtimeExecutable` with "tsx" to run .ts file as tsx
